servicebus/spring-cloud-azure-starter-integration-servicebus/single-namespace/README.md

```diff
@@ -1,183 +1,89 @@
-
-1.  Update [application-sp.yaml][application-sp.yaml].
-    ```yaml
-    spring:
-      cloud:
-        azure:
-          credential:
-            client-id: ${AZURE_CLIENT_ID}
-            client-secret: ${AZURE_CLIENT_SECRET}
-          profile:
-            tenant-id: ${AZURE_TENANT_ID}
-          servicebus:
-            namespace: ${AZURE_SERVICEBUS_BINDER_NAMESPACE}
-    ```
-    > We should specify `spring.profiles.active=sp` to run the Spring Boot application.
-    For App Service, please add a configuration entry for this.
-#### Method 3: MSI credential based usage
-
-##### Set up managed identity
-
-Please follow [create managed identity][create-managed-identity] to set up managed identity.
-
-##### Add Role Assignment for Service Bus
-
-1.  See [Managed identities for Azure resources with Service Bus][role-assignment]
-    to add role assignment for Service Bus. Assign `Contributor` role for managed identity.
+variable "service_principal_id" {
+  type        = string
+  description = "The Azure Service Principal id or object_id to assign role to"
+  default     = ""
+}
+```
 
+2. Use the terraform script to create Azure Resources
+```shell
 
-##### Update MSI related properties
+cd terraform
 
-1.  Update [application-mi.yaml][application-mi.yaml].
-    ```yaml
-    spring:
-      cloud:
-        azure:
-          credential:
-            managed-identity-client-id: ${AZURE_MANAGED_IDENTITY_CLIENT_ID}
-          profile:
-            tenant-id: ${AZURE_TENANT_ID}
-          servicebus:
-            namespace: ${AZURE_SERVICEBUS_NAMESPACE}
-    ```
-    > We should specify `spring.profiles.active=mi` to run the Spring Boot application.
-    For App Service, please add a configuration entry for this.
+# Login to azure-cli
+az login
+az account set --subscription <your-subscription-id>
 
-##### Redeploy Application
+# 
+terraform init
+# it may take a few minutes
+terraform apply
+```
 
-If you update the `spring.cloud.azure.credential.managed-identity-client-id`
-property after deploying the app, or update the role assignment for
-services, please try to redeploy the app again.
 
-> You can follow
-> [Deploy a Spring Boot JAR file to Azure App Service][deploy-spring-boot-application-to-app-service]
-> to deploy this application to App Service
+## Run locally
 
-## Examples
+Open your terminal, and input the command below.
 
-1. Run the `mvn spring-boot:run` in the root of the code sample to get the app running.
+```shell
+mvn clean spring-boot:run
+```
 
-2. Update queue name in 
-   [QueueReceiveController.java][queue-receive-controller] and
-   [QueueSendController.java][queue-send-controller],
-   and update topic name and subscription in
-   [TopicReceiveController.java][topic-receive-controller] and
-   [TopicSendController.java][topic-send-controller].   
-    
-3. Run the `mvn spring-boot:run` in the root of the code sample to get the app running.
 
-4. Send a POST request to service bus queue
+## Run on App Service
```

Comment:
it should be run on azure spring cloud, not app service, since azure spring cloud is the best place to run spring boots application